src/Stratis.Bitcoin/Consensus/Validators/BlockValidator.cs

```diff
@@ -19,8 +19,9 @@ public interface IHeaderValidator
         /// Validates a block header.
         /// </summary>
         /// <param name="chainedHeader">The chained header to be validated.</param>
+        /// <param name="blockMined">Whether the block was mined or obtained from a peer.</param>
         /// <returns>Context that contains validation result related information.</returns>
-        ValidationContext ValidateHeader(ChainedHeader chainedHeader);
+        ValidationContext ValidateHeader(ChainedHeader chainedHeader, bool blockMined);
```

Comment:
Do we need this?

Reply:
Can we not solve this in a different way?
this is so ugly.

Reply:
I'm open for suggestions. 

Basically we need this thing to be populated for all types of validation so the only way I found without having a refactoring is to provide this bool in all validators. That leaded to propagation of this bool from high level code to the very bottom.  

Another solution would be to remove `BlockMined` from context and all the rules should treat the stuff being validated as something downloaded from the peer. I'd actually go with that but it's a bit big refactoring.